### PR TITLE
AAD support for WPS and new API from swagger to generate token

### PR DESCRIPTION
**Pull request by srnagar**
Moving the PR https://github.com/Azure/azure-sdk-for-java/pull/23861 to release branch to allow pipelines to run.

### CODE REVIEW

File: sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/WebPubSubServiceAsyncClient.java [hunk 1]
```diff
@@ -128,10 +179,10 @@ public WebPubSubAuthenticationToken getAuthenticationToken(GetAuthenticationToke
      * Close the client connection.
      * @param connectionId Target connection Id.
      * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @return the completion.
```
Comment (JonathanGiles) [inline sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/WebPubSubServiceAsyncClient.java:182]: ugh

File: sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/ReverseProxyPolicy.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.CoreUtils;
+import reactor.core.publisher.Mono;
+
+import java.net.URL;
+
+/**
+ * The reverse proxy policy.
+ */
+public final class ReverseProxyPolicy implements HttpPipelinePolicy {
```
[the remark this answers is not in the document]
Comment (srnagar) [reply sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/ReverseProxyPolicy.java:19]: Yes, this most likely belongs in azure-core. .NET currently has this policy in WPS library. I'll check with other languages and get approval from other architects to move to core.

File: sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/WebPubSubServiceAsyncClient.java (hunk 1 above)
Comment (srnagar) [reply sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/WebPubSubServiceAsyncClient.java:182]: Need better javadoc in LLC autorest codegen. Filed an issue for autorest - https://github.com/Azure/autorest.java/issues/1178

**Review by JonathanGiles — APPROVED** [review]

**Pull request merged**